packages/core/src/lib/register.ts

```diff
@@ -97,3 +98,32 @@ export const registerWithEmailAndPasscode = async (
 
   await assignRegistrationResult(ctx, provider, id);
 };
+
+export const registerWithSocial = async (
+  ctx: WithUserLogContext<Context>,
+  provider: Provider,
+  { connectorId, code }: { connectorId: string; code: string }
+) => {
+  const userInfo = await getUserInfoByCode(connectorId, code);
```

Comment:
nit: would it be more precise to rename it as `getUserInfoByConnectorCode`?

Reply:
this 'code' field is reused and represents different types of codes: in email and SMS cases, it refers to `passcode` and refers to `authorization code` in these social connector cases. not sure whether 'ConnectorCode' is a proper naming, maybe `getUserInfoByAuthorizationCode` is a more clear name.

Reply:
name it as `ConnectorCode` can prevent the confusing of email and SMS, so let's keep it short.